? 639618_notified_warning_mysql.patch
? 774826_use_defaults_if_user_has_none.patch
? _comment_notify.underscore_in_content_type.patch.txt
? unfinished_hash_update.patch
Index: comment_notify.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/comment_notify/comment_notify.module,v
retrieving revision 1.73.2.5
diff -u -p -r1.73.2.5 comment_notify.module
--- comment_notify.module	9 Jul 2010 15:01:39 -0000	1.73.2.5
+++ comment_notify.module	11 Oct 2010 01:00:04 -0000
@@ -340,6 +340,11 @@ function comment_notify_user($type, &$ed
           $user->$property = $value;
         }
       }
+      else {
+        // If they have no preference, use the defaults.
+        $user->node_notify_mailalert = variable_get('node_notify_default_mailalert', FALSE);
+        $user->comment_notify_mailalert = variable_get('comment_notify_default_registered_mailalert', COMMENT_NOTIFY_NODE);
+      }
       break;
 
     case 'delete':
